src/storage/store/var_list_column_chunk.cpp

```diff
@@ -86,14 +86,25 @@ void VarListColumnChunk::append(common::ValueVector* vector, common::offset_t st
         if (vector->isNull(pos)) {
             continue;
         }
-        auto listEntry = vector->getValue<list_entry_t>(pos);
-        dataVector->state->selVector->selectedSize = listEntry.size;
-        for (auto j = 0u; j < listEntry.size; j++) {
-            dataVector->state->selVector->selectedPositions[j] = listEntry.offset + j;
+        copyListValues(vector->getValue<list_entry_t>(pos), dataVector);
+    }
+    numValues += vector->state->selVector->selectedSize;
+}
+
+void VarListColumnChunk::copyListValues(const list_entry_t& entry, ValueVector* dataVector) {
+    auto numListValuesToCopy = entry.size;
+    auto numListValuesCopied = 0;
+    while (numListValuesCopied < numListValuesToCopy) {
+        auto numListValuesToCopyInBatch =
+            std::min<uint64_t>(numListValuesToCopy - numListValuesCopied, DEFAULT_VECTOR_CAPACITY);
+        dataVector->state->selVector->selectedSize = numListValuesToCopyInBatch;
+        for (auto j = 0u; j < numListValuesToCopyInBatch; j++) {
+            dataVector->state->selVector->selectedPositions[j] =
+                entry.offset + numListValuesCopied + j;
         }
         varListDataColumnChunk.append(dataVector);
```

Comment:
Why does this code work, reusing the same vector?

Reply:
The selectionVector can only store the state for 2048 values, we should only append 2048 values at a time when the list is large.

Reply:
Yeah. My actual confusion was I didn't realize `varlistdatacolumnChunk.append()` copies the valuevector